SQLQueryGroupProjectTeamB1Julia.sql

```diff
@@ -0,0 +1,255 @@
+				JOIN tblEMP_SHIFT_STATUS ESS
+					ON S.ShiftID = ESS.ShiftID
+				JOIN tblEMPLOYEE_POSITION EP
+					ON ESS.EmpPosID = EP.EmpPosID
+				JOIN tblEMPLOYEE E
+					ON EP.EmployeeID = E.EmployeeID
+				JOIN tblDAY D
+					ON D.DayID = S.DayID
+				JOIN tblMONTH M
+					ON M.MonthID = S.MonthID
+			  WHERE DAY(E.DOB) = D.[DayName]
+				AND MONTH(E.DOB) = M.MonthNum
+			 )
+	BEGIN
+			SET @RET = 1
+	END
+	RETURN @RET
+END
+GO
+
+ALTER TABLE tblSHIFT
+ADD CONSTRAINT BR_NoShiftsOnBirthday
+CHECK(dbo.BR_NoShiftsOnEmpBirthday( ) = 0)
+GO
+
+--Enforce the business rule preventing multiple shifts of the same day/time to be scheduled - one employee per shift
+
+CREATE FUNCTION dbo.BR_NoScheduleShiftsAtSameTime()
+RETURNS INT
+AS 
+BEGIN
+	DECLARE @RET INT = 0
+	IF EXISTS(SELECT ESS.ESSID, COUNT(ESS.EmpPosID)
+			  FROM tblEMP_SHIFT_STATUS ESS
+				JOIN tblSHIFT S
+					ON ESS.ShiftID = S.ShiftID
+				JOIN tblQUARTER Q
+					ON Q.QuarterID = S.QuarterID
+				JOIN tblMONTH M
+					ON M.MonthID = S.MonthID
+				JOIN tblDAY D
+					ON D.DayID = S.DayID
+				JOIN tblHOUR H
+					ON H.HourID = S.BeginHourID
+				JOIN tblHOUR H2
+					ON H2.HourID = S.EndHourID
+				JOIN tblLOCATION L
+					ON L.LocationID = S.LocationID
+			  GROUP BY S.ShiftID, S.QuarterID, S.MonthID, S.DayID, S.BeginHourID, S.EndHourID, S.LocationID
+			  HAVING COUNT(ESS.EmpPosID) > 1
+			)
+	BEGIN
+		SET @RET = 1
+	END
+	RETURN @RET
+END
+GO
+
+ALTER TABLE tblSHIFT
+ADD CONSTRAINT BR_NoShiftsSameTime
+CHECK(dbo.BR_NoScheduleShiftsAtSameTime( ) = 0)
+GO
+
+--COMPUTED COLUMNS
+
+--Write the user-defined function enabling the computed column showing the number of hours an employee has worked within the past 7 days.
+
+CREATE FUNCTION FN_TotalHoursForEmpInCurrentMonth(@PK INT)
+RETURNS NUMERIC(4,2)
+AS 
+BEGIN
+	DECLARE @RET INT = (SELECT SUM(S.DurationHours)
+					    FROM tblEMPLOYEE E
+							JOIN tblEMPLOYEE_POSITION EP
+								ON E.EmployeeID = EP.EmployeeID
+							JOIN tblEMP_SHIFT_STATUS ESP
+								ON EP.EmpPosID = ESP.EmpPosID
+							JOIN tblSHIFT S
+								ON ESP.ShiftID = S.ShiftID
+							JOIN tblMONTH M
+								ON M.MonthID = S.MonthID
+						WHERE E.EmployeeID = @PK
+							AND M.[MonthNum] = Month(GetDate())
+						)
+	RETURN @RET
+END
+GO
+
+ALTER TABLE tblEMPLOYEE
+ADD TotalHoursInCurrentMonth AS (dbo.FN_TotalHoursForEmpInCurrentMonth(EmployeeID))
+GO
+
+--Write the user-defined function enabling the computed column number of employees on each location
+
+CREATE FUNCTION FN_NumEmployeesPerLocation(@PK INT)
+RETURNS NUMERIC(6,2)
+AS 
+BEGIN
+	DECLARE @RET INT = (SELECT L.LocationName, SUM(E.EmployeeID) AS NumEmployees
+					    FROM tblEMPLOYEE E
+							JOIN tblEMPLOYEE_POSITION EP
+								ON E.EmployeeID = EP.EmployeeID
+							JOIN tblEMP_SHIFT_STATUS EPS
+								ON EP.EmpPosID = EPS.EmpPosID
+							JOIN tblSHIFT S
+								ON EPS.ShiftID = S.ShiftID
+							JOIN tblLOCATION L
+								ON S.LocationID = L.LocationID
+						GROUP BY L.LocationName

```

Comment:
For Computed columns, you don't want to have a group by, you are filtering your location through the passed Location ID which will be filtered in the WHERE clause